app/portainer/components/template-list/template-list-controller.js

```diff
@@ -0,0 +1,56 @@
+angular.module('portainer.app').controller('TemplateListController', ['DatatableService',
+  function TemplateListController(DatatableService) {
+    var ctrl = this;
+
+    this.state = {
+      textFilter: '',
+      selectedCategory: '',
+      categories: [],
+      showContainerTemplates: true
+    };
+
+    function onTextFilterChange() {
```

Comment:
You are not exporting `onTextFilterChange()` as a part of the component API. This is probably why the filter is not persisted on update.

```suggestion
    this.onTextFilterChange() {
```